CONTRIBUTING.md

```diff
@@ -1,60 +1,81 @@
 
-Please let us know via our GitHub issue tracker if you find a problem, even if you don't have a fix for it.
-The ideal issue report should be descriptive, and where possible include the steps we can take to reproduce the problem for ourselves.
+## Reporting Issues
+Issues are tracked using [Jira](https://issues.redhat.com/secure/RapidBoard.jspa?rapidView=18680#). If you encounter a bug or have suggestions for enhancements, please follow the steps below:
 
-Please go to [issue tracker](https://github.com/opendatahub-io/opendatahub-operator/issues) and create a "New issue".
-Choose suitable issue type and fill in description as detailed as possible.
-Lastly, add extra label for this issue if that can help us refine it.
+1. **Check for Existing Issues:** Before creating a new issue, search the Jira project to see if a similar issue already exists.
+2. **Create a Jira Ticket:** If the issue doesn’t exist, create a new ticket in Jira. 
+   - **For Feature Requests:**  Set the issue type to be `Initative`
+   - **For Bugs:** Set the issue type to `Bug`
+   - **For all other code changes:** Use the issue type `Story`
 
-If you have a proposed fix for an issue, or an enhancement you would like to make to the code please describe it in an issue, then send us the code as a GitHub pull request as described below.
+## Pull Requests
 
-## Pull request
+### Workflow
 
-Use a descriptive title, and if it relates to an issue in our tracker please reference which one.
-If the PR is not intended to be merged you should prefix the title with "[WIP]" which indicates it is still Work In Progress.
-PR's description should provide enough information for a reviewer to understand the changes and their relation to the rest of the code.
+1. **Fork the Repository:** Create your own fork of the repository to work on your changes.
+2. **Create a Branch:** Create your own branch to include changes for the feature or a bug fix off of `incubation` branch.
+3. **Work on Your Changes:** Commit often, and ensure your code adheres to these [Code Style Guidelines](#code-style-guidelines) and passes all the [quality gates](#quality-gates) for the operator.
+4. **Testing:** Make sure your code passes all the tests, including any new tests you've added. 
 
-## Setting up a Fedora-based development environment
+### Open a Pull Request:
 
-This is a loose list of tools to install on your linux box in order to compile, test and deploy the operator. 
+1. **Link to Jira Issue**: Include the Jira issue link in your PR description.
+2. **Description**: Provide a detailed description of the changes and what they fix or implement.
+3. **Add Testing Steps**: Provide information on how the PR has been tested, and list out testing steps if any for reviewers.
+4. **Review Request**: Tag the relevant maintainers(@opendatahub-io/odh-operator-maintainers ) or team members(@opendatahub-io/odh-platform-members) for a review.
```

Comment:
We can mention the [PR template](https://github.com/opendatahub-io/opendatahub-operator/blob/15dfe1ae7e117e614232b511eb1236f2af6e5049/.github/PULL_REQUEST_TEMPLATE.md) here as well.